### DISCUSSION TITLE
How exactly does renovate work with Gitlab

### DISCUSSION THREAD

**Discussion by orange-puff** [renovatebot/renovate#11190] [Request Help] [answered]
So I understand that to setup renovate with Gitlab, I invoke the renovate command line tool in a Gitlab pipeline, it reads the config for repos to scan and creates PRs for any packages that have fallen behind in those repos, but I don't understand exactly how it searches the repos for packages. Does it clone them and then search them locally or does it use the Gitlab API?

**Comment by viceice** [comment, marked as answer]
it clones the repo's with git and searhes and updates locally
